File: LeetCode/230KthSmallestElementInABST/KthSmallestElementInABST.cpp
class Solution {
public:
    int ans = -1;
    int index = 1;
    void inorder(TreeNode* root, int k) {
        if (!root) {
            return;
        }
        inorder(root->left, k);
        if (k == index) {
            ans = root->val;
        }
        index++;
        inorder(root->right, k);
    }
    int kthSmallest(TreeNode* root, int k) {
        // DFS -> Time:O(n), Space:O(h)
        inorder(root, k);
        return ans;
    }
};

File: LeetCode/230KthSmallestElementInABST/KthSmallestElementInABST.py
class Solution:
    index = 1
    ans = -1
    def inorder(self, root, k):
        if not root:
            return
        self.inorder(root.left, k)
        if k == self.index:
            self.ans = root.val
        self.index += 1
        self.inorder(root.right, k)

    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        # DFS -> Time:O(n), Space:O(h)
        self.inorder(root, k)
        return self.ans
